### DISCUSSION TITLE
[V5] Trying to do a kind of groupBy (Need help)

### DISCUSSION THREAD

**Discussion by Reptiluka** [adonisjs/core#1894] [Help] [answered]
Hi !
In my database, I have two row like that (example)
Table : Example
{ custom_id : 1 , name: "Test1", second_info: "Test1info"}
{ custom_id : 1 , name: "Test2", second_info: "Test2info"}

**Yes, the two rows have the same custom_id**

If i do : `let example = await Example.query().groupBy('custom_id')` it will only keep 1 row

How I can and what is the best way using query builder to get that structure? : 
[ 1 : [ { custom_id : 1 , name: "Test1", second_info: "Test1info"}{ custom_id : 1 , name: "Test2", second_info: "Test2info"} ] ]
So,
[ the custom id : [ An array of results from all rows corresponding the custom id ]

Thats looks weird and simple at the same time, but I need help,
Thanks ! :)

**Comment by steschwa** [comment, marked as answer]
Just load the required columns with a query like `const query = await Example.query().where("custom_id", 1)`.
Because you get a typed `Example[]` value back it's possible to utilize all the available array methods. To produce your desired output:

```ts
const example = query.reduce(
    (prev, next) => ({
        ...prev,
        [next.custom_id]: [...(prev[next.custom_id] ?? []), next ]
    }), {})
```

**Reply by Reptiluka** [reply]
It works, thats all I wanted, THANK YOU ! :D
